tmol/tests/score/common/test_uaid_util.py

```diff
@@ -0,0 +1,398 @@
+    block_type_atom_downstream_of_conn[0, 1, 0] = 2
+    block_type_atom_downstream_of_conn[0, 1, 1] = 1
+    block_type_atom_downstream_of_conn[0, 1, 2] = 0
+
+    resolved = resolve_uaids(
+        uaids,
+        block_inds,
+        pose_inds,
+        pose_stack_block_coord_offset,
+        pose_stack_block_type,
+        pose_stack_inter_block_connections,
+        block_type_atom_downstream_of_conn,
+    )
+
+    resolved_gold = torch.tensor([-1, -1], dtype=torch.int32)
+    numpy.testing.assert_equal(resolved_gold.numpy(), resolved.numpy())
+
+
+def test_resolve_unspecified_uaids():
+    n_blocks = 6
+    n_uaids = 2
+    uaids = torch.full((n_uaids, 3), -1, dtype=torch.int32)
+    block_inds = torch.zeros((n_uaids,), dtype=torch.int32)
+    pose_inds = torch.zeros((n_uaids,), dtype=torch.int32)
+    pose_stack_block_coord_offset = torch.zeros((1, n_blocks), dtype=torch.int32)
+    pose_stack_block_type = torch.zeros((1, n_blocks), dtype=torch.int32)
+    pose_stack_inter_block_connections = torch.full(
+        (1, n_blocks, 2, 2), -1, dtype=torch.int32
+    )
+    block_type_atom_downstream_of_conn = torch.zeros((1, 2, 20), dtype=torch.int32)
+
+    block_inds[0] = 1
+    block_inds[1] = 2
+
+    pose_stack_block_coord_offset[0, 0] = 0
+    pose_stack_block_coord_offset[0, 1] = 20
+    pose_stack_block_coord_offset[0, 2] = 40
+    pose_stack_block_coord_offset[0, 3] = 60
+    pose_stack_block_coord_offset[0, 4] = 80
+    pose_stack_block_coord_offset[0, 5] = 100
+
+    # first residue
+    pose_stack_inter_block_connections[0, 0, 0, 0] = -1
+    pose_stack_inter_block_connections[0, 0, 0, 1] = -1
+    pose_stack_inter_block_connections[0, 0, 1, 0] = 1
+    pose_stack_inter_block_connections[0, 0, 1, 1] = 0
+    # 2nd residue
+    pose_stack_inter_block_connections[0, 1, 0, 0] = 0
+    pose_stack_inter_block_connections[0, 1, 0, 1] = 1
+    pose_stack_inter_block_connections[0, 1, 1, 0] = 2
+    pose_stack_inter_block_connections[0, 1, 1, 1] = 0
+    # 3rd residue
+    pose_stack_inter_block_connections[0, 2, 0, 0] = 1
+    pose_stack_inter_block_connections[0, 2, 0, 1] = 1
+    pose_stack_inter_block_connections[0, 2, 1, 0] = 3
+    pose_stack_inter_block_connections[0, 2, 1, 1] = 0
+    # 4th residue
+    pose_stack_inter_block_connections[0, 3, 0, 0] = 2
+    pose_stack_inter_block_connections[0, 3, 0, 1] = 1
+    pose_stack_inter_block_connections[0, 3, 1, 0] = 4
+    pose_stack_inter_block_connections[0, 3, 1, 1] = 0
+    # 5th residue
+    pose_stack_inter_block_connections[0, 4, 0, 0] = 3
+    pose_stack_inter_block_connections[0, 4, 0, 1] = 1
+    pose_stack_inter_block_connections[0, 4, 1, 0] = 5
+    pose_stack_inter_block_connections[0, 4, 1, 1] = 0
+    # 6th residue
+    pose_stack_inter_block_connections[0, 5, 0, 0] = 4
+    pose_stack_inter_block_connections[0, 5, 0, 1] = 1
+    pose_stack_inter_block_connections[0, 5, 1, 0] = -1
+    pose_stack_inter_block_connections[0, 5, 1, 1] = -1
```

Comment:
Seems like a lot of these definitions could be shared between the different tests.

Reply:
yes, created a fixture